scripts/ci/build_libtiledb.sh

```diff
@@ -23,11 +23,12 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.
 #
+set -euo pipefail
```

Comment:
Let's remove `u` for now because we have an unbound variable in `scripts/ci/build_benchmarks.sh` that the CI is now complaining about.

Reply:
I fixed that, but it looks like there is an actual test failure on macOS + S3: https://github.com/TileDB-Inc/TileDB/pull/2252/checks?check_run_id=2527902739
```
2021-05-07T14:11:03.5587540Z ===============================================================================
2021-05-07T14:11:03.5587950Z test cases:     599 |     598 passed | 1 failed
2021-05-07T14:11:03.5588410Z assertions: 6492761 | 6492758 passed | 3 failed
```